iothub/service/src/Registry/Models/Device.cs

```diff
@@ -35,80 +34,79 @@ public Device(string id)
         /// <summary>
         /// Device Id.
         /// </summary>
-        [JsonProperty(PropertyName = "deviceId")]
+        [JsonProperty("deviceId")]
         public string Id { get; internal set; }
 
         /// <summary>
-        /// Device's Generation Id.
+        /// Device's generation Id.
         /// </summary>
         /// <remarks>
         /// This value is used to distinguish devices with the same deviceId, when they have been deleted and re-created.
         /// </remarks>
-        [JsonProperty(PropertyName = "generationId")]
+        [JsonProperty("generationId")]
         public string GenerationId { get; internal set; }
 
         /// <summary>
         /// Device's ETag.
         /// </summary>
-        [JsonProperty(PropertyName = "etag")]
-        [JsonConverter(typeof(NewtonsoftJsonETagConverter))] // NewtonsoftJsonETagConverter is used here because otherwise the ETag isn't serialized properly
+        [JsonProperty("etag")]
+        // NewtonsoftJsonETagConverter is used here because otherwise the ETag isn't serialized properly
+        [JsonConverter(typeof(NewtonsoftJsonETagConverter))]
         public ETag ETag { get; set; }
 
         /// <summary>
         /// Device's connection state.
         /// </summary>
-        [JsonProperty(PropertyName = "connectionState")]
-        [JsonConverter(typeof(StringEnumConverter))]
+        [JsonProperty("connectionState")]
         public ClientConnectionState ConnectionState { get; internal set; }
 
         /// <summary>
         /// Device's status.
         /// </summary>
-        [JsonProperty(PropertyName = "status")]
-        [JsonConverter(typeof(StringEnumConverter))]
+        [JsonProperty("status")]
         public ClientStatus Status { get; set; }
 
         /// <summary>
         /// Reason, if any, for the device to be in specified status.
         /// </summary>
-        [JsonProperty(PropertyName = "statusReason")]
+        [JsonProperty("statusReason")]
         public string StatusReason { get; set; }
 
         /// <summary>
         /// Time when the connection state was last updated.
         /// </summary>
-        [JsonProperty(PropertyName = "connectionStateUpdatedTime")]
-        public DateTimeOffset ConnectionStateUpdatedOnUtc { get; internal set; }
+        [JsonProperty("connectionStateUpdatedTime")]
+        public DateTimeOffset? ConnectionStateUpdatedOnUtc { get; internal set; }
 
         /// <summary>
         /// Time when the status was last updated.
         /// </summary>
-        [JsonProperty(PropertyName = "statusUpdatedTime")]
-        public DateTimeOffset StatusUpdatedOnUtc { get; internal set; }
+        [JsonProperty("statusUpdatedTime")]
+        public DateTimeOffset? StatusUpdatedOnUtc { get; internal set; }
```

Comment:
Nullable so when someone creates a device, it doesn't have Jan1 0001 as a default value.